components/org.wso2.carbon.stream.processor.common/src/main/java/org/wso2/carbon/stream/processor/common/utils/config/FileConfigManager.java

```diff
@@ -89,6 +89,29 @@ public ConfigReader generateConfigReader(String namespace, String name) {
             } catch (ConfigurationException e) {
                 LOGGER.error("Could not initiate the siddhi configuration object, " + e.getMessage(), e);
             }
+
+            if (name.equalsIgnoreCase("shardId")) {
+                try {
+                    ClusterConfiguration clusterConfig =
+                            configProvider.getConfigurationObject(ClusterConfiguration.class);
+                    if (clusterConfig != null) {
+                        if (clusterConfig.getGroupId() != null && clusterConfig.isEnabled()) {
+                            Map<String, String> configs = new HashMap<>();
```

Comment:
Initiate map object only once, at the start of the function and return after adding the needed configs